### PR TITLE
Add support for AVRA linting

### PR DESCRIPTION
**Pull request by UtkarshVerma**
This PR adds support for AVR assembly linting through [`avra`](https://github.com/Ro5bert/avra).

This is the linter in action:
Peek 2021-10-21 07-43

### CODE REVIEW

File: test/linter/test_avra_avra.vader [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+  let b:ale_avra_avra_executable = '~/avra'
+
+  AssertLinter '~/avra', ale#Escape('~/avra') . b:command_tail
+
+Execute(The options should be configurable):
+  let b:ale_avra_avra_options = '--max_errors 20'
+
+  AssertLinter 'avra', ale#Escape('avra')
+  \ . ' --max_errors 20 %t -o ' . g:ale#util#nul_file
```
[the remark this answers is not in the document]
Comment (UtkarshVerma) [reply test/linter/test_avra_avra.vader]: I reordered the arguments. The tests are working now.

**Comment by UtkarshVerma** [comment]
Also, if you're merging this PR, could you please consider labelling this PR as 'hacktoberfest-accepted'?

File: test/handler/test_avra_handler.vader [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  \     'type': 'E',
+  \   },
+  \   {
+  \     'lnum': 12,
+  \     'text': "Unknown directive: .EQ",
+  \     'type': 'E',
+  \   },
+  \ ],
+  \ ale_linters#avra#avra#Handle(bufnr(''), [
+  \   "main.asm(3) : Error   : Unknown device: atmega3228p"
```
Comment (hsanson) [inline test/handler/test_avra_handler.vader:22]: Tests are failing due to missing commas "," after each item in this list.

Comment (UtkarshVerma) [reply test/handler/test_avra_handler.vader]: I've fixed the regex. Tests should work now

**Comment by UtkarshVerma** [comment]
@hsanson I have fixed the tests. Everything should be fine now.

**Comment by UtkarshVerma** [comment]
@hsanson I have fixed the tests. Everything should be fine now.

**Comment by hsanson** [comment]
Make sure to add and align these two documents:

 -  supported-tools.md and doc/ale-supported-languages-and-tools.txt 

Tests fail because the new linter seems to be missing or out of order in one of them.

**Review by hsanson — APPROVED** [review]
Thanks for the changes.

**Pull request merged**